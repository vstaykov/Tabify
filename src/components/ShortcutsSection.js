"use strict";

import React from "react";
import Shortcut from "./Shortcut.js"

const ShortcutsSection = (props) => {

    return (
        <div>
            <div>Shortcuts Cheatsheet</div>
            <hr />
            {props.shortcuts.map((shortcut) => {
                return (
                    <Shortcut key={shortcut.keys} {...shortcut} />
                );
            })}
        </div>
    );
}

export default ShortcutsSection;