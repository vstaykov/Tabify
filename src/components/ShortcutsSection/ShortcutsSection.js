import React from "react";
import PropTypes from "prop-types";
import Shortcut from "../Shortcut/Shortcut";

const ShortcutsSection = ({ shortcuts }) => (
  <div className="shortcuts-section">
    <div className="shortcuts-section-title">Shortcuts Cheatsheet</div>
    <hr />
    {shortcuts.map(shortcut => <Shortcut {...shortcut} key={shortcut.title} />)}
  </div>
);

ShortcutsSection.propTypes = {
  shortcuts: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      keys: PropTypes.arrayOf(PropTypes.string).isRequired
    })
  ).isRequired
};

export default ShortcutsSection;
