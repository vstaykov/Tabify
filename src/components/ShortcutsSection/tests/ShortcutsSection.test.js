import React from "react";
import { shallow } from "enzyme";
import ShortcutsSection from "../ShortcutsSection";

describe("<ShortcutsSection />", () => {
  describe("render()", () => {
    describe("when given one shortcut", () => {
      it("render shortcut correctly", () => {
        const props = {
          shortcuts: [
            {
              title: "Foo",
              keys: ["Ctrl"]
            }
          ]
        };

        const component = shallow(<ShortcutsSection {...props} />);

        expect(component).toMatchSnapshot();
      });
    });

    describe("when given several shortcuts", () => {
      it("render all shortcuts correctly", () => {
        const props = {
          shortcuts: [
            {
              title: "Foo",
              keys: ["Ctrl"]
            },
            {
              title: "Bar",
              keys: ["Alt", "Q"]
            },
            {
              title: "FooBar",
              keys: ["Ctrl", "Shift", "Alt", "Q"]
            }
          ]
        };

        const component = shallow(<ShortcutsSection {...props} />);

        expect(component).toMatchSnapshot();
      });
    });
  });
});
