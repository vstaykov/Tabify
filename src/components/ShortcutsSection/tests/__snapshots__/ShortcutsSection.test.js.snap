// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShortcutsSection /> render() when given one shortcut render shortcut correctly 1`] = `
<div
  className="shortcuts-section"
>
  <div
    className="shortcuts-section-title"
  >
    Shortcuts Cheatsheet
  </div>
  <hr />
  <Shortcut
    key="Foo"
    keys={
      Array [
        "Ctrl",
      ]
    }
    title="Foo"
  />
</div>
`;

exports[`<ShortcutsSection /> render() when given several shortcuts render all shortcuts correctly 1`] = `
<div
  className="shortcuts-section"
>
  <div
    className="shortcuts-section-title"
  >
    Shortcuts Cheatsheet
  </div>
  <hr />
  <Shortcut
    key="Foo"
    keys={
      Array [
        "Ctrl",
      ]
    }
    title="Foo"
  />
  <Shortcut
    key="Bar"
    keys={
      Array [
        "Alt",
        "Q",
      ]
    }
    title="Bar"
  />
  <Shortcut
    key="FooBar"
    keys={
      Array [
        "Ctrl",
        "Shift",
        "Alt",
        "Q",
      ]
    }
    title="FooBar"
  />
</div>
`;
